Document `generateClasses` (#2633)
cap-js/docs: java/developing-applications/building.md

  The method `getService()` in generated [event-specific Event Context interfaces](../event-handlers/#eventcontext) is overridden to return the typed service interface instead of the generic `Service` type.

- [`generateClasses`](../assets/cds-maven-plugin-site/generate-mojo.html#generateClasses)

  Generates static class-based implementations for interfaces, replacing dynamic proxies at runtime.

>   This improves performance and removes the limit on the number of methods an interface may declare.

Reviewer: ```suggestion
  This improves the performance significantly. It also removes the limit on the number of methods an interface may declare.
```

>   This improves the performance significantly. It also removes the limit on the number of methods an interface may declare.

Reviewer: ```suggestion
  This improves the performance significantly. It also removes the limit on the number of methods an accessor interface may declare.
```

>   This improves the performance significantly. It also removes the limit on the number of methods an accessor interface may declare.

Reviewer 2: ```suggestion
  This significantly improves performance and removes the limit on the number of methods an accessor interface may declare.
```

>   The generated classes are internal — do not reference them directly in application code.

Reviewer: ```suggestion
  The generated classes are internal — do not reference them directly from application code.
```

>   Interfaces not produced by the code generator are still implemented dynamically, falling back to proxies when no static implementation exists.

Reviewer: ```suggestion
  Hand-written [accessor interfaces](../cds-data#typed-access), for which no implementation class is generated, are still supported by falling back to using [proxies](../cds-data#cds-data).
```

>   Hand-written [accessor interfaces](../cds-data#typed-access), for which no implementation class is generated, are still supported by falling back to using [proxies](../cds-data#cds-data).

Reviewer 2: ```suggestion
  Hand-written [accessor interfaces](../cds-data#typed-access), for which no implementation class is generated, are still supported via [proxy](../cds-data#cds-data) fallback.
```
